isis/src/clipper/objs/ClipperPushBroomCamera/ClipperPushBroomCamera.cpp

```diff
@@ -44,10 +44,14 @@ namespace Isis {
 
      NaifStatus::CheckErrors();
 
-     SetFocalLength();
+     Pvl &lab = *cube.label();
+
+     PvlGroup &bandBin = lab.findGroup("BandBin", Pvl::Traverse);
+     QString key = "INS" + toString(naifIkCode()) + "_" + bandBin["FilterName"][0] + "_FOCAL_LENGTH";
+     SetFocalLength(Spice::getDouble(key));
```

Comment:
Ended up not placing this SetFocalLength call in a try/catch because getDouble throws adequate and detailed error statements. For example if the key was not in the BandBin, getDouble outputs `The camera is requesting spice data [INS-159103_CLEAR_FOCAL_LENGTH] that was not attached, please re-run spiceinit." thrown in the test body.`